Src/IronPython/Runtime/ByteArray.cs

```diff
@@ -301,11 +301,15 @@ public ByteArray center(int width, [BytesLike, NotNull]IList<byte> fillchar)
             }
         }
 
-        public bool endswith([BytesLike, NotNull]IList<byte> suffix, object? start = null, object? end = null) {
+        public bool endswith([BytesLike, NotNull]IList<byte> suffix, object? start) {
+            return endswith(suffix, start, null);
+        }
+
+        public bool endswith([BytesLike, NotNull]IList<byte> suffix, object? start, object? end) {
             int istart = start != null ? Converter.ConvertToIndex(start) : 0;
             lock (this) {
                 int iend = end != null ? Converter.ConvertToIndex(end) : _bytes.Count;
-                return _bytes.StartsWith(suffix, istart, iend);
+                return _bytes.EndsWith(suffix, istart, iend);
```

Comment:
Didn't even notice that it was incorrectly using `StartsWith`. This overload is not covered by a test?

Reply:
It is, but the test is ambiguous. Testing if _abc_ starts with _abc_ works as well as testing if it ends with _abc_. I will adapt the tests to catch that. Apparently starring at the code for hours just doesn't cut it... :wink:

I am also locally using `test_bytes.py` from StdLib, but it did not catch it either. It did catch some other things though. To further extend our test coverage, I will check in `test_bytes_stdlib.py` that selectively runs those tests form StdLib that currently pass. This file can be deleted once `test_bytes.py` is enabled. This is similar to what I did for codecs.